### PR TITLE
fix(cli): dispatch unload event on watch drop

### PR DESCRIPTION
**Pull request by caspervonb**
Fixes https://github.com/denoland/deno/issues/7452

### CODE REVIEW

File: cli/main.rs
Comment (bartlomieju) [inline cli/main.rs:906]: This is a good solution, but please add some docstrings here - it's not immediately obvious why this is needed. I'd also suggest a change of name, maybe `FileWatcherModuleExecutor`?

Comment (caspervonb) [reply cli/main.rs:906]: > FileWatcherModuleExecutor

As for the naming, wasn't the original intent but I'm thinking this can be a more generic thing.
Think it might be useful to hoist this in a follow up and use it everywhere where we do this lifecycle sequence which typically looks like:
- maybe_start_coverage_collection
- execute_module
- dispatch `load`
- run_event_loop
- dispatch 'unload'
- maybe_stop_coverage_collection

We repeat these steps in a few places that do the typical execution cycle.
But I'm leaving that for a (somewhat immediate) follow-up, first pass I'm just about fixing this bug.

**Comment by bartlomieju** [comment]
@caspervonb please rebase, I think we'll land it for 1.13.2

**Review by bartlomieju — APPROVED** [review]
LGTM, I'll wait for Luca's review before landing

**Review by lucacasonato — APPROVED** [review]

**Pull request merged**
